core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/LDAPConnectionContext.java

```diff
@@ -203,10 +209,87 @@ public LDAPConnectionContext(RealmConfiguration realmConfig) throws UserStoreExc
         // Set StartTLS option if provided in the configuration. Otherwise normal connection.
         startTLSEnabled = Boolean.parseBoolean(realmConfig.getUserStoreProperty(
                 UserStoreConfigConstants.STARTTLS_ENABLED));
+
+        // Set waiting time to re-establish LDAP connection after couple of failure attempts if specified otherwise
+        // set default time.
+        String retryWaitingTime = realmConfig.getUserStoreProperty(UserStoreConfigConstants.CONNECTION_RETRY_DELAY);
+        if (StringUtils.isNotEmpty(retryWaitingTime)) {
+            thresholdTimeoutInMilliseconds = getThresholdTimeoutInMilliseconds(retryWaitingTime);
+        } else {
+            thresholdTimeoutInMilliseconds = 120000;
```

Comment:
can we make this number a constant and user here as well https://github.com/wso2/carbon-kernel/pull/2260/files#r323552420

Reply:
Fixed with https://github.com/wso2/carbon-kernel/pull/2272